src/Eloquent/Collection.php

```diff
@@ -49,4 +49,59 @@ public function toTree($childrenRelation = 'children')
 
         return $tree;
     }
+
+    /**
+     * Load parent/anscestor relations already present in the tree.
+     *
+     * @return static<int, TModel>
+     */
+    public function loadTreePathRelations()
+    {
+        $instance = $this->first();
+
+        if (is_null($instance)) {
+            return $this;
+        }
+
+        if (! method_exists($instance, 'getPathName') || ! method_exists($instance, 'getPathSeparator')) {
+            throw new RuntimeException(sprintf(
```

Comment:
Please import the `RuntimeException` class.

Reply:
Done.